infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -333,4 +340,8 @@ private String getAMIParameterValue(InfrastructureConfig infrastructureConfig)
             throws TestGridInfrastructureException {
         return amiMapper.getAMIFor(infrastructureConfig.getParameters());
     }
+
+    private String getOSParameterValue(InfrastructureConfig infrastructureConfig) {
+        return infrastructureConfig.getParameters().getProperty("OS");
```

Comment:
operating_system is already a parameter. OS name and version can be properties of operating_system, yes/

Reply:
Where do we provide operating_system as a parameter to CF?

Reply:
It is a generic infra param that is read from the infrastructure_parameter table.

Reply:
AFAIK we do not provide operating_system as a parameter to CF since it is only used to lookup the AMI. Also current CF script does not have a parameter for OS.

Reply:
Yes, that's the current usage. But, a CF can request for operating_system if it requires. This is just another infrastructure value..

Reply:
What do you propose?

Reply:
Like the 'database' infrastructure parameter has DBEngine and DBEngineVersion as properties, the 'operating_system' should have 'OSName' and 'OSVersion'

Reply:
WDYT?

Reply:
+1. That should be added to the relevant db table so that it would be picked when generating test-plans. However, we need to fetch values from the infrastructure-config in the generated test-plan.

Reply:
Cool.. I didn't understand your last sentence. :-) The infrastructure_parameter table's properties column already get processed in the generated-testplan. 

[1] https://github.com/wso2/testgrid/pull/619/files#diff-46d3c40a2de643e74df9e250c4642fa0R297

Reply:
Yes. But we only provide properties and values that are defined inside the scripts/inputParameters section to scripts. Those input-parameters are taken from the relevant infra.yaml file. So even-though the test-plan already has the OS property, it won't be used on CF. Hence we need a way to feed OS name into the CF script.

Reply:
And, we shouldn't specialize 'OS' value; testgrid should treat it as just another parameter.